src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -206,3 +220,67 @@ private[hyperspace] object IndexSummary {
     indexDirPath.toString
   }
 }
+
+/**
```

Comment:
@Reviewers:
Using a case class is a clean/easy way to create a DataFrame with proper schema.
`IndexStatistics` includes the members of `IndexSummary`. Ideally it should have extended `IndexSummary`, however AFAIK Case classes in Scala do not support inheritance. I am open to hear your suggestions to avoid duplicating `IndexSummary` members here.

Reply:
I'm okay with the approach as it looks more concise.

Reply:
Have we thought about using "extended" similar to `DESCRIBE TABLE EXTENDED` or `SHOW TABLE EXTENDED`?

For example, we have the following in `Hyperspace`:
```
def indexes: DataFrame
```

We can maybe update it as the following:
```
def indexes(extended: Boolean = false): DataFrame

def index(extended: Boolean = false): DataFrame
```

In that way, we can have one structure and pick short version if `extended` is false.

Reply:
I thought about this a bit more and this is what I am proposing:
* The APIs can be similar to `SHOW TABLES` and `SHOW TABLE EXTENDED` as follows:
```scala
def indexes: DataFrame // same as existing (behavior is similar to "SHOW TABLES")

def index(name: String): DataFrame // just def name change from "getIndexStats" (behavior is similar to "SHOW TABLE EXTENDED)
```

* Have one structure (`IndexSummary`) to represent the index info:
For `indexes`, we can select fewer fields when we create the dataframe. For `index`, we will select all the fields by default ("extended" behavior). If we need to groups "stats", we can create `IndexStats` and make it as a member of `IndexSummary`; it will create nested fields, but it's fine. But, let's avoid duplicating fields across different case classes.

WDYT, @pirz @sezruby ?

Reply:
How about using only `IndexStats` and filter additional columns for `indexes`? 
I think it's better than using nested fields. e.g.
for indexes: `Seq(IndexStatistics(spark, toIndexLogEntry(l), extended=false)).toDF().select(name, indexedColumns, includedColumns ...`
for index: `Seq(IndexStatistics(spark, toIndexLogEntry(l), extended=true)).toDF()`

For indexes & index naming, it seems better for consistency :)

Reply:
Yes, either way is fine as long as we don't duplicate fields (and nested fields/flatten only if we want to group fields which may not be necessary at this point)

Reply:
@imback82 I had thought about using the same structure for both commands and selecting fewer fields when creating DF for `indexes`. My concern was performance:
1. Fields like `sizeIndexFiles` and `sizeSourceFiles` can take some time to calculate if the index has thousands of files or sources. But with `indexes` we calculate them and then exclude them from the output (i.e. waste of effort).
2. `indexes` fetches information for "all" indexes, so calculating expensive fields for all indexes and truncating them is just not elegant.

Regarding your suggestion to consolidate structures, are you suggesting switching from a "case class" to a normal class and then adding functions to convert an instance of class to a DF with proper schema?

Reply:
You can avoid the overhead with a new arg(extended) in apply function, for 'indexes'.

Case class should work

Reply:
This is done - `IndexSummary` is removed and a single case class `IndexStatistics` is used for both APIs: `indexes(..)` and `index(..)`.